vmdb/spec/lib/miq_automation_engine/miq_ae_state_machine_new_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'spec_helper'
+require 'timecop'
+
+describe "MiqAeStateMachine" do
+  before do
+    DummyWorkspace = Class.new do
+      def initialize(options = {})
+        @options = options
+      end
+
+      def root
+        @options
+      end
+    end
+
+    TestClass = Class.new do
+      include MiqAeEngine::MiqAeStateMachine
+      def initialize(options = {})
+        @workspace = DummyWorkspace.new(options)
+      end
+    end
+  end
+
+  after do
+    Object.send(:remove_const, :TestClass)
+    Object.send(:remove_const, :DummyWorkspace)
+  end
+
+  context "enforce_max_retries" do
+    it "exceeds retry count" do
+      obj = TestClass.new('ae_state_retries' => 3)
+      expect { obj.enforce_max_retries('max_retries' => 2) }.to raise_error
```

Comment:
could you say what error you are expecting to raise?
```ruby
expect {}.to raise_error(/time in state/)
```

Is there code somewhere that will catch this? Will they want to know the exact error raised?
Would it make sense to name the error with a class like `MiqAeEngine::TimeoutError` or something?
Not sure if we already have them. I'm guessing you can get away with a single exception for all these use cases, unless you see the calling code needing to distinguish between these various cases.

If an error is an error, maybe a generic automate error that already exists would be good enough.